vault/audited_headers.go

```diff
@@ -120,8 +131,15 @@ func (c *Core) setupAuditedHeadersConfig() error {
 		}
 	}
 
+	// Ensure that we are able to case-sensitively access the headers;
```

Comment:
This is so low impact that I don't see the point in ever deprecating it. No reason to take that chance that someone will go three versions without upgrading and then fall afoul of us removing the upgrade path.